openlibrary/core/bookshelves_events.py

```diff
@@ -124,6 +125,25 @@ def select_distinct_by_user_type_and_year(cls, username, event_type, year):
 
         return list(oldb.query(query, vars=data))
 
+    @classmethod
+    def total_yearly_reading_goals(cls, since: date | None = None) -> int:
+        """Returns (int) number of reading goals updated. `since` may be used
+        to limit the result to those books logged since a specific date. Any
+        python datetime.date type should work.
+
+        :param since: returns all logged books after date
+        """
+        oldb = db.get_db()
+
+        query = "SELECT count(*) from yearly_reading_goals"
+        if since:
+            query += " WHERE updated >= $since"
+        results = cast(
+            tuple[int],
+            oldb.query(query, vars={'since': since}),
+        )
+        return results[0]
```

Comment:
```suggestion
        results = tuple(oldb.query(query, vars={'since': since}))
        return int(results[0])
```
[`typing.cast()` is a way to blind typecheckers](https://adamj.eu/tech/2021/07/06/python-type-hints-how-to-use-typing-cast/)  like [mypy](https://mypy-lang.org) but at runtime it is a noop...
> At runtime, cast() really does do nothing - its special behavior is only in how type checkers interpret it. [Python’s source code](https://github.com/python/cpython/blob/f64de53ff01e734d48d1d42195443d7d1646f220/Lib/typing.py#L1628-L1636) reveals cast() is a simple no-op function call.
```
>>> cast(tuple[int], ("A", "B", "C"))
('A', 'B', 'C')
```
The call to `typing.cast()` does not convert the strings to ints.  It just blinds the type checker from seeing that they are not ints.

We want to make sure that this function returns an `int` to its caller so let's use an equivalent form.
```
>>> cast(tuple[int], ("A", "B", "C")) == tuple(("A", "B", "C"))
True
```
And then let's explicitly call `int()` on the return value so that any type errors will be raised in this function and we do not return bad data to the caller.
```
>>> results = tuple(("A", "B", "C"))
>>> results
('A', 'B', 'C')
>>> int(results[0])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: invalid literal for int() with base 10: 'A'
```